enumerable.rb

```diff
@@ -0,0 +1,186 @@
+module Enumerable
+  def my_each
+    if self.class == Range
+      l = to_a
+      count = 0
+      while count < l.length
+        yield(l[count])
+        count += 1
+      end
+      l
+    elsif self.class == Hash
+      count = 0
+      while count < length
+        yield(self[count])
+        count += 1
+      end
+      self
+    else
+      count = 0
+      while count < length
+        yield(self[count])
+        count += 1
+      end
+      self
+    end
+  end
+
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+
+  def my_select
+    array = []
+    my_each do |i|
+      array.push(i) if yield(i)
+    end
+    array
+  end
+
+  def my_all?
+    my_each do |i|
+      (return true if yield(i) == true)
+      (return false if yield(i) == false)
+    end
+    true
+  end
+
+  def my_any?
+    count = 0
+    state = false
+    my_each do |i|
+      count += 1 if yield(i) == true
+      count += 0 if yield(i) == false
+    end
+    state = true if count.positive?
+    state = false if count.zero?
+    state
+  end
+
+  def my_none?
+    count = 0
+    state = false
+    my_each do |i|
+      count += 1 if yield(i) == true
+      count += 0 if yield(i) == false
+    end
+    state = false if count.positive?
+    state = true if count.zero?
+    state
+  end
+
+  def my_count
```

Comment:
* [ ] should have working `my_count` **(covered by automated tests)** identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
``` ar = [1, 2, 4, 2] ```
``` p ar.my_count(2) != ar.count(2) ```